testsuite/features/step_definitions/vm_steps.rb

```diff
@@ -0,0 +1,432 @@
+# Copyright (c) 2023 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+### This file contains all steps concerning virtual machine management
+
+When(/^I do a late hostname initialization of host "([^"]*)"$/) do |host|
+  # special handling for e.g. nested VMs since they will only be crated later on
+  # this step is normally done int twopence_init.rb
+  node = get_target(host)
+
+  hostname, _local, _remote, code = node.test_and_store_results_together('hostname', 'root', 500)
+  raise "Empty hostname for #{host}" if hostname.empty?
+
+  raise "Cannot connect to get hostname for '#{host}'. Response code: #{code}" if code.nonzero?
+  raise "No hostname for '#{host}'. Response code: #{code}" if hostname.empty?
+
+  node.init_hostname(hostname)
+  fqdn, _local, _remote, code = node.test_and_store_results_together('hostname -f', 'root', 500)
+
+  raise "Cannot connect to get FQDN for '#{host}'. Response code: #{code}" if code.nonzero?
+  raise "No FQDN for '#{host}'. Response code: #{code}" if fqdn.empty?
+
+  node.init_full_hostname(fqdn)
+end
```

Comment:
Personally I would move this step to another Ruby file.
The reason is that this is not a special case only for Nested VMs.
I temporary need the same lines when I was testing the Pod Proxy while it was initialized later inside the Proxy VM.
I made it not necessary doing some tricks. But this could be useful for other cases aside of nested VMs.

Reply:
Yes, makes sense. I moved it to the `command_steps.rb` file.